packages/ember-metal/lib/mixin.js

```diff
@@ -809,7 +809,7 @@ export function observer(...args) {
 
   if (typeof func !== 'function') {
     // revert to old, soft-deprecated argument ordering
-    Ember.deprecate('Passing the dependentKeys after the callback function in Ember.observer is deprecated. Ensure the callback function is the last argument.');
+    Ember.deprecate('Passing the dependentKeys after the callback function in Ember.observer is deprecated. Ensure the callback function is the last argument.', false, { id: 'ember-metal.observer', until: '3.0.0' });
```

Comment:
Let's make this one 'ember-metal.observer-argument-order'. I can imagine a future where we would want to deprecate or change Ember.observer itself...